db/mysql/dao/gateway.go

```diff
@@ -313,6 +313,66 @@ func (h *HTTPRuleDaoImpl) ListByComponentIDs(componentIDs []string) ([]*model.HT
 	return rules, nil
 }
 
+// HTTPRuleRewriteDaoTmpl is a implementation of HTTPRuleRewriteDao
+type HTTPRuleRewriteDaoTmpl struct {
+	DB *gorm.DB
+}
+
+//AddModel -
+func (h *HTTPRuleRewriteDaoTmpl) AddModel(mo model.Interface) error {
+	httpRuleRewrite, ok := mo.(*model.HTTPRuleRewrite)
+	if !ok {
+		return fmt.Errorf("can't not convert %s to *model.HTTPRuleRewrite", reflect.TypeOf(mo).String())
+	}
+	var oldHTTPRuleRewrite model.HTTPRuleRewrite
+	if ok := h.DB.Where("uuid = ?", httpRuleRewrite.UUID).Find(&oldHTTPRuleRewrite).RecordNotFound(); ok {
+		if err := h.DB.Create(httpRuleRewrite).Error; err != nil {
+			return err
+		}
+	} else {
```

Comment:
This condition branch is not necessary. You can directly return the existing error of HTTPRuleRewrite